.github/workflows/pypi-publish.yml

```diff
@@ -0,0 +1,33 @@
+name: Publish Python Package
+
+on:
+  release:
+    types: 
+        - published
+
+permissions:
+    contents: read    
+
+jobs:
+  pypi-publish:
+    name: upload release to PyPI
+    runs-on: ubuntu-latest
+    permissions:
+      # This permission is needed for private repositories.
+      contents: read
+      # IMPORTANT: this permission is mandatory for trusted publishing
+      id-token: write
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.release.tag_name }}
+          
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+            python-version: '3.9'
+      - uses: pdm-project/setup-pdm@v3
+      
```

Comment:
@jaseemjaskp Better if we leave a single empty line only above and below this.

Reply:
@hari-kuriakose this only for testing. I will remove the dry run and update it in next PR. I can't able to test it without merging into main.